bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/AbstractWatchService.java

```diff
@@ -77,7 +85,9 @@ protected void initializeWatchService() {
                 if (watchSubDirectories()) {
                     watchService = FileSystems.getDefault().newWatchService();
 
-                    Files.walkFileTree(toWatch, new SimpleFileVisitor<Path>() {
+                    // walk through all folders and follow symlinks
+                    Files.walkFileTree(toWatch, Sets.newHashSet(FileVisitOption.FOLLOW_LINKS), MAX_DEPTH, 
```

Comment:
`Files.walkFileTree(toWatch, new SimpleFileVisitor<Path>() {...});` has been used before.

The JavaDoc you linked states:
```text
public static Path walkFileTree(Path start,
                                FileVisitor<? super Path> visitor)
                         throws IOException
Walks a file tree.
This method works as if invoking it were equivalent to evaluating the expression:

 walkFileTree(start, EnumSet.noneOf(FileVisitOption.class), Integer.MAX_VALUE, visitor)
 
In other words, it does not follow symbolic links, and visits all levels of the file tree.
```

So, if you want to follow symbolic links, too I would prefer to change only that parameter and keep the other defaults.

If we also want to be able to change the depth, we should do it in a separate PR.

> @maggu2810: Do you prefer using Integer.MAX_VALUE

Yes :wink: